fix(compiler): inputs/outputs incorrectly parsed in jit mode

crisbeto opened the pull request:
The `Directive` and `Component` decorators support `inputs` and `outputs` fields which accept an array in the format of `"someInput"` or `"someInput: someAlias"`, however the parsing during JIT compilation was splitting on commas, not on colons, which resulted in incorrect parsing. E.g. `inputs: ["someInput: someAlias"]` was being parsed into `{"someInput: someAlias": "someInput: someAlias"}` instead of `{someInput: "someAlias"}`.

The feature was working by accident, because there's some logic further down in the compiler pipeline that was splitting the strings again.

crisbeto commented:
[Passing TGP](http://test/OCL:460642752:BASE:460686737:1657714137207:567b1117).

AndrewKushnir commented on packages/compiler/src/render3/view/util.ts:
```diff
@@ -186,12 +186,9 @@ function mapToExpression(
-      // Otherwise the declared name should be omitted even if it is different from the public name,
-      // as it may have already been minified.
-      needsDeclaredName = publicName !== declaredName && key.includes(':');
+      minifiedName = declaredName = key;
```
There is one small comment at the beginning of the function that talks about `:`. The comment becomes obsolete and we can drop it (or update it to describe the current state).

AndrewKushnir reviewed (approved):
@crisbeto the change looks great 👍

I've found the original PR where the code (that splits by comma) was introduced: #24565. It looks like there are no special use-cases that we are missing, so it should be safe to proceed with this change.

jessicajaniuk reviewed (approved):
LGTM :cookie:

crisbeto committed:
fix(compiler): inputs/outputs incorrectly parsed in jit mode

The `Directive` and `Component` decorators support `inputs` and `outputs` fields which accept an array in the format of `"someInput"` or `"someInput: someAlias"`, however the parsing during JIT compilation was splitting on commas, not on colons, which resulted in incorrect parsing. E.g. `inputs: ["someInput: someAlias"]` was being parsed into `{"someInput: someAlias": "someInput: someAlias"}` instead of `{someInput: "someAlias"}`.

The feature was working by accident, because there's some logic further down in the compiler pipeline that was splitting the strings again.

crisbeto commented:
Pushed a change to address Andrew's feedback. It may need re-approval.

jessicajaniuk commented:
This PR was merged into the repository by commit f67d2cabaf2bce235c693e0716fb1f167ba8a2c7.
